cpp/memilio/mobility/graph.h

```diff
@@ -305,88 +305,73 @@ class Graph
-          class ReadFunction, class NodeIdFunction>
-IOResult<void> set_nodes(const Parameters& params, Date start_date, Date end_date, const fs::path& data_dir,
-                         const std::string& population_data_path, bool is_node_for_county,
-                         Graph<Model, MobilityParams>& params_graph, ReadFunction&& read_func,
-                         NodeIdFunction&& node_func, const std::vector<FP>& scaling_factor_inf, FP scaling_factor_icu,
-                         FP tnt_capacity_factor, int num_days = 0, bool export_time_series = false,
-                         bool rki_age_groups = true)
-
+template <class Model, class TestAndTrace>
+void set_test_and_trace_capacity(const mio::VectorRange<Node<Model>>& nodes, ScalarType tnt_capacity_factor)
 {
-    BOOST_OUTCOME_TRY(auto&& node_ids, node_func(population_data_path, is_node_for_county, rki_age_groups));
-    std::vector<Model> nodes(node_ids.size(), Model(int(size_t(params.get_num_groups()))));
-    for (auto& node : nodes) {
-        node.parameters = params;
-    }
-
-    BOOST_OUTCOME_TRY(read_func(nodes, start_date, node_ids, scaling_factor_inf, scaling_factor_icu, data_dir.string(),
-                                num_days, export_time_series));
-
     for (size_t node_idx = 0; node_idx < nodes.size(); ++node_idx) {
 
-        auto tnt_capacity = nodes[node_idx].populations.get_total() * tnt_capacity_factor;
+        auto tnt_capacity = nodes[node_idx].property.populations.get_total() * tnt_capacity_factor;
 
-        //local parameters
-        auto& tnt_value = nodes[node_idx].parameters.template get<TestAndTrace>();
-        tnt_value       = UncertainValue<FP>(0.5 * (1.2 * tnt_capacity + 0.8 * tnt_capacity));
+        auto& tnt_value = nodes[node_idx].property.parameters.template get<TestAndTrace>();
+        tnt_value       = UncertainValue<ScalarType>(tnt_capacity);
         tnt_value.set_distribution(mio::ParameterDistributionUniform(0.8 * tnt_capacity, 1.2 * tnt_capacity));
+    }
+}
 
-        auto id = 0;
-        if (is_node_for_county) {
-            id = int(regions::CountyId(node_ids[node_idx]));
-        }
-        else {
-            id = int(regions::DistrictId(node_ids[node_idx]));
-        }
-        //holiday periods
-        auto holiday_periods = regions::get_holidays(regions::get_state_id(id), start_date, end_date);
-        auto& contacts       = nodes[node_idx].parameters.template get<ContactPattern>();
+/**
+ * @brief Set german state holidays for the given nodes.
+ * 
+ * Works for nodes of a graph depicting german counties or states.
+ * 
+ * @param[in, out] nodes VectorRange of Node%s each containing a Model in which the data is set.
+ * @param[in] start_date Date at the beginning of the simulation.
+ * @param[in] end_date Date at the end of the simulation.
+ */
+template <class FP, class Model, class ContactPattern>
+void set_german_holidays(const mio::VectorRange<Node<Model>>& nodes, const mio::Date& start_date,
```

Comment:
in generell, we should try to keep this generic and add a overall function set_holidays which allows a country specifier e.g., for Germany.
however, since we only have data for germany at this moment, this is ok :)